docs/how-tos/provision.md

```diff
@@ -0,0 +1,310 @@
+
+```sh {class="command-line" data-prompt="$"}
+wget https://storage.googleapis.com/packages.viam.com/apps/viam-agent/preinstall.sh
+chmod 755 preinstall.sh
+```
+
+{{< /tablestep >}}
+{{< tablestep >}}
+**2. Run the script**
+
+Run the preinstall script without options and it will attempt to auto-detect a mounted root filesystem (or for Raspberry Pi, bootfs) and also automatically determine the architecture.
+
+```sh {class="command-line" data-prompt="$"}
+sudo ./preinstall.sh
+```
+
+If this succeeds, continue to step 5.
+Alternatively, go to step 3.
+
+{{< /tablestep >}}
+{{< tablestep >}}
+**3. Run the script for the target system's architecture**
+
+The following command creates a tarball you can extract on your own:
+
+{{< tabs >}}
+{{% tab name="arm64" %}}
+
+```sh {class="command-line" data-prompt="$"}
+sudo ./preinstall.sh --aarch64
+```
+
+{{% /tab %}}
+{{% tab name="x86_64" %}}
+
+```sh {class="command-line" data-prompt="$"}
+sudo ./preinstall.sh --x86_64
+```
+
+{{% /tab %}}
+{{< /tabs >}}
+
+If this command fails because the script cannot detect your mountpoint, you can specify it directly:
+
+```sh {class="command-line" data-prompt="$"}
+sudo ./preinstall.sh /path/to/rootfs
+```
+
+{{< /tablestep >}}
+{{< /table >}}
+
+{{% /tab %}}
+{{% tab name="Preinstall on an SD card" %}}
+
+{{< table >}}
+{{< tablestep >}}
+**1. Flash a 64-bit image to your SD card**
+
+Flash a 64-bit image to your SD card using the Raspberry Pi Imager, and customize at least the hostname when prompted.
+
+{{< /tablestep >}}
+{{< tablestep >}}
+**2. Run the preinstall script**
+
+Eject and reinsert the card to make sure it's mounted with the newly written contents. Then run the following script:
+
+```sh {class="command-line" data-prompt="$"}
+sudo /bin/sh -c "$(curl -fsSL https://storage.googleapis.com/packages.viam.com/apps/viam-agent/preinstall.sh)"
+```
+
+Follow the instructions:
+
+```sh {class="command-line" data-prompt="$" data-output="2-40"}
+coding sudo /bin/sh -c "$(curl -fsSL https://storage.googleapis.com/packages.viam.com/apps/viam-agent/preinstall.sh)"
+
+
+Found Raspberry Pi bootfs mounted at /Volumes/bootfs
+
+
+A Raspberry Pi boot partition has been found mounted at /Volumes/bootfs
+This script will modify firstrun.sh on that partition to install Viam agent.
+Continue pre-install? (y/n): y
+Path to custom viam-provisioning.json (leave empty to skip):
+Creating tarball for install.
+a opt
+a opt/viam
+a opt/viam/cache
+a opt/viam/bin
+a opt/viam/bin/viam-agent
+a opt/viam/bin/agent-provisioning
+a opt/viam/cache/viam-agent-provisioning-factory-aarch64
+a opt/viam/cache/viam-agent-factory-aarch64
+a etc
+a usr
+a usr/local
+a usr/local/lib
+a usr/local/lib/systemd
+a usr/local/lib/systemd/system
+a usr/local/lib/systemd/system/viam-agent.service
+a usr/local/lib/systemd/system/multi-user.target.wants
+a usr/local/lib/systemd/system/multi-user.target.wants/viam-agent.service
+
+
+Install complete! You can eject/unmount and boot the image now.
```

Comment:
If I turn the machine on, it doesn't start the hotspot.

Reply:
do you have machine credentials  in the machine?

Reply:
no, neither. I thought if I don't configure internet then it should go into hotspot mode, and that should then request machine credentials

Reply:
@Otterverse is it required that I have a working network on the device already as well as a `viam.json`? I thought the whole point was that if I didn't a hotspot would start... I guess I'll continue testing tomorrow. If we have steps for this written down anywhere else, please let me know.

Reply:
You're correct. viam-provisioning.json is ALWAYS optional (think of it as "branding/customization" for the manufacturer) whereas viam.json is the machine credentails, and that IS exactly what provisioning is meant to help set up. So it's absolutely valid for it to be missing if you want to have a machine do provisioning at first boot.

If you pre-installed the agent on the SD card, then it should absolutely start up a hotspot and allow you to provide both network and machine credentials from there. It can take several minutes for first boot, but if after 5+ minutes nothing is happening, then something may have gone wrong. Feel free to slack me if you want some help troubleshooting.

Reply:
I do think we should hide the password on the captive portal. It's a bit inconvenient as is